### ISSUE TITLE
windows incremental compile is broken?

### ISSUE THREAD

**Issue by lovettchris** [microsoft/react-native-windows#11880]
### Problem Description

Sorry for the dumb question, but I can't get past a silly compile error after adding a simple KeyDown handler to the default `npx react-native-windows-init --overwrite` app MainPage.  I added this to MainPage.xml:
```xml
    KeyDown="MainKeyDown"
```
and this to the MainPage.h:

```cpp
        void MainKeyDown(
            Windows::Foundation::IInspectable const& sender, 
            Windows::UI::Xaml::Input::KeyRoutedEventArgs const& e);
```

and this to MainPage.cpp:
```cpp
    void MainPage::MainKeyDown(
        Windows::Foundation::IInspectable const& /*sender*/, Windows::UI::Xaml::Input::KeyRoutedEventArgs const& e)
    {
        auto key = e.Key();
        if (key == winrt::Windows::System::VirtualKey::F11)
        {
            e.Handled(true);
        }
    }
```
But the ReactNativeApp.sln in VS 2022 reports compile error on the first line there saying:

> error C3536: 'key': cannot be used before it is initialized

What am I missing?

PS: After doing a full clean rebuild, the compile error went away??? So perhaps  incremental build is broken in this case?

### Steps To Reproduce

See above

### Expected Results

incremental compile should work.

### CLI version

npx react-native -v

### Environment

```markdown
npx react-native info

11.3.5
```

### Target Platform Version

None

### Visual Studio Version

None

### Build Configuration

None

**Comment by chrisglein** [comment]
Yes, may be broken. But first, what version are you on? (run the environment scripts above)

**Comment by lovettchris** [comment]
Sorry about that, I added the missing info and tried again from scratch, then I found the solution which is to add these missing includes:

```
#include <winrt/Windows.System.h>           // define VirtualKey
#include <winrt/Windows.UI.Xaml.Input.h> // define KeyRoutedEventArgs
#include <winrt/Windows.Foundation.h>    // define IInspectable
```

So the problem is VS intellisense is too smart and was showing my all types and stuff, (even F12 was working on all those types) like everything was nicely defined, when it actually wasn't...

**Comment by lovettchris** [comment]
Pity the error message doesn't say "Cannot find types IInspectable, KeyRoutedEventArgs and VirtualKey".

**Comment by jonthysell** [comment]
I see, yes it would be nice if VS identified the missing headers you were using.
